### PR TITLE
[jamendo] parsing of duration (closes #18565)

### LINKED ISSUE #18565: Jamendo "WARNING: unable to extract duration"
### Checklist

- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.12.17**
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### Output

```
$ youtube-dl -v https://www.jamendo.com/track/592180/winter-dreams
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.jamendo.com/track/592180/winter-dreams']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.12.09
[debug] Python version 3.7.1 (CPython) - Darwin-18.2.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.1, ffprobe 4.1
[debug] Proxy map: {}
[Jamendo] winter-dreams: Downloading webpage
WARNING: unable to extract duration; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on 'https://flac.jamendo.com/?trackid=592180&format=flac&from=app-97dab294'
[download] Vladimir Sterzer - Winter Dreams-592180.flac has already been downloaded
[download] 100% of 12.62MiB
```

```
$ youtube-dl -v -g https://www.jamendo.com/track/592180/winter-dreams
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-g', 'https://www.jamendo.com/track/592180/winter-dreams']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.12.09
[debug] Python version 3.7.1 (CPython) - Darwin-18.2.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.1, ffprobe 4.1
[debug] Proxy map: {}
WARNING: unable to extract duration; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
[debug] Default format spec: bestvideo+bestaudio/best
https://flac.jamendo.com/?trackid=592180&format=flac&from=app-97dab294
```

### Description of your *issue*, suggested solution and other information

`youtube-dl` is warning me that it can't find a duration. It still manages to download the file and show its URL.

### PR DESCRIPTION
**Pull request by matthias4217**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Youtube-dl hasn't been able to fetch jamendo song duration for a while (#18565). This should fix it. I'm not totally happy with my fix, as I've encountered trouble fetching it from a json, and had to use it in another place in the page (see my comment on the issue. However, tests work fine (I've added a new one by the way).

### CODE REVIEW

File: youtube_dl/extractor/jamendo.py [hunk 1]
```diff
@@ -35,6 +35,19 @@ class JamendoIE(JamendoBaseIE):
...
+            'id': '1311140',
+            'display_id': '221',
+            'ext': 'flac',
+            'title': 'Duration of life - 221',
+            'artist': 'Duration of life',
+            'track': '221',
+            'duration': 141,
+            'thumbnail': r're:^https?://.*\.jpg'
+        }
+    }, {
```
Comment (dstftw) [inline youtube_dl/extractor/jamendo.py]: Remove duplicate tests.

File: youtube_dl/extractor/jamendo.py [hunk 2]
```diff
@@ -80,7 +93,7 @@ def _real_extract(self, url):
         thumbnail = self._html_search_meta(
             'image', webpage, 'thumbnail', fatal=False)
         duration = parse_duration(self._search_regex(
-            r'<span[^>]+itemprop=["\']duration["\'][^>]+content=["\'](.+?)["\']',
+            r'data-bundled-models=["\']{[^}]+{[^{*}]+duration&quot;:(.+?),',
```
Comment (dstftw) [inline youtube_dl/extractor/jamendo.py]: No, parse as JSON. Or from ld+json.

File: youtube_dl/extractor/jamendo.py (hunk 1 above)
Comment (matthias4217) [reply youtube_dl/extractor/jamendo.py]: I'll push it when I've fixed the duration extraction.

File: youtube_dl/extractor/jamendo.py (hunk 2 above)
Comment (matthias4217) [reply youtube_dl/extractor/jamendo.py]: Thank you for your review ! I've tried to do that, however I guess that Jamendo has an issue with the song duration in the ld+json part. Time format for the duration is ISO8601 I guess, but for one song lasting 210s, it is written as `PT0.21S` whereas for another lasting 141s, it is `PT0.141S`. Thus, parsing the Json would give incoherent results I guess. Moreover, this format can't be parsed by `utils.parse8601()`, so I don't really know what to do.

**Pull request closed without merging**